Applications/ApplicationsLib/TestDefinition.cpp

```diff
@@ -163,27 +166,43 @@ TestDefinition::TestDefinition(BaseLib::ConfigTree const& config_tree,
             vtkdiff_config.getConfigParameter<std::string>("field");
         DBUG("vtkdiff will compare field '{:s}'.", field_name);
 
+        std::vector<std::string> filenames;
+        if (auto const regex_string =
+            //! \ogs_file_param{prj__test_definition__vtkdiff__regex}
+            vtkdiff_config.getConfigParameterOptional<std::string>("regex"))
+        {
+            // TODO: insert rank into regex for mpi case
+            DBUG("vtkdiff regex is '{}'.", *regex_string);
+            auto const regex = std::regex(*regex_string);
+            for (auto& p: fs::directory_iterator(fs::path(reference_path)))
+            {
+                auto const filename = p.path().filename().string();
+                if (std::regex_match(filename, regex))
+                {
+                    DBUG("        -> matched '{}'", filename);
+                    filenames.push_back(filename);
+                }
+            }
+        }
+        else
+        {
 #ifdef USE_PETSC
-        int rank;
-        MPI_Comm_rank(PETSC_COMM_WORLD, &rank);
-        int mpi_size;
-        MPI_Comm_size(PETSC_COMM_WORLD, &mpi_size);
-        std::string const& filename =
-            MeshLib::IO::getVtuFileNameForPetscOutputWithoutExtension(
-                //! \ogs_file_param{prj__test_definition__vtkdiff__file}
-                vtkdiff_config.getConfigParameter<std::string>("file")) +
-            "_" + std::to_string(rank) + ".vtu";
+            int rank;
+            MPI_Comm_rank(PETSC_COMM_WORLD, &rank);
+            int mpi_size;
+            MPI_Comm_size(PETSC_COMM_WORLD, &mpi_size);
+            std::string const& filename =
```

Comment:
I don't understand why filename should be a reference. The referenced object on the right hand side is constructed.

Reply:
Why not. The lifetime of the returned temporary (if it is a temporary) is extended to the lifetime of the reference. The copy happens in line 204...
The `getVtuFileName...()` could return a reference to something local.